DataManagementSystem/Utilities/CatalogUtilities.py

```diff
@@ -0,0 +1,140 @@
+########################################################################
+# $HeadURL$
+########################################################################
+
+""" DIRAC FileCatalog utilities
+"""
+
+__RCSID__ = "$Id$"
+
+import hashlib as md5
+
+import random
+import os
+import errno
+
+from DIRAC import S_OK
+from DIRAC.Core.Utilities.DErrno import DError
+
+def checkArgumentFormat( path, generateMap = False ):
+  """ Bring the various possible form of arguments to FileCatalog methods to
+      the standard dictionary form
+  """
+
+  def checkArgumentDict( path ):
+    """ Check and process format of the arguments to FileCatalog methods """
+    if isinstance( path, basestring ):
+      urls = {path:True}
+    elif isinstance( path, list ):
+      urls = {}
+      for url in path:
+        urls[url] = True
+    elif isinstance( path, dict ):
+      urls = path
+    else:
+      return DError( errno.EINVAL, "Utils.checkArgumentFormat: Supplied path is not of the correct format." )
+    return S_OK( urls )
+
+  if not path:
+    return DError( errno.EINVAL, 'Empty input: %s' % str( path ) )
+
+  result = checkArgumentDict( path )
+  if not result['OK']:
+    return result
+
+  pathDict = result['Value']
+
+  # Bring the lfn path to the normalized form
+  urls = {}
+  urlMap = {}
+  for url in pathDict:
+    # avoid empty path...
+    if not url:
+      continue
+    mUrl = url
+    if url.startswith( 'lfn:' ):
+      mUrl = url[4:]
+    elif url.startswith( 'LFN:' ):
+      mUrl = url[4:]
+    if mUrl.startswith('/grid'):
+      uList = mUrl.split('/')
+      if len( uList ) >= 2 and uList[1] == 'grid':
+        mUrl = mUrl[5:]
```

Comment:
yes, but if we could manipulate LFN paths like paths and not like strings it may be clearer, right? unless we want to run on Windows where os.path.sep is not '/' of course...